synthetic_data/generators.py

```diff
@@ -49,23 +75,163 @@ def post_profile_processing_w_data(cls, data, profile):
             )
         return profile
 
-    def synthesize(self, num_samples: int, seed=None, noise_level: float = None):
+    def synthesize(
+        self,
+        num_samples: int,
+        noise_level: float = None,
+    ):
+        """Generate synthetic tabular data."""
+        if noise_level is None:
+            noise_level = self.noise_level
+
+        if self.is_correlated:
+            return make_data_from_report(
+                report=self.profile.report(),
```

Comment:
Might be another PR, but how does make_data_from_report determine the distribution of "possible" discrete columns. One of the columns was the iris target (0/1) --- synthetic data produced 0.0, 1.0, 2.0

Reply:
follow-up PR / creating an issue for this and below. Need to think about this